neo/rawio/baserawio.py

```diff
@@ -12,34 +12,50 @@
 When possible, all IOs should/implement this level following these guidelines:
   * internal use of memmap (or hdf5)
   * fast reading of the header (do not read the complete file)
-  * neo tree object is symetric and logical: same channel/units/event
+  * neo tree object is symmetric and logical: same channel/units/event
     along all block and segments.
 
-
-So this handles **only** one simplified but very frequent case of dataset:
-    * Only one channel set  for AnalogSignal (aka ChannelIndex) stable along Segment
-    * Only one channel set  for SpikeTrain (aka Unit) stable along Segment
-    * AnalogSignal have all the same sampling_rate acroos all Segment
-    * t_start/t_stop are the same for many object (SpikeTrain, Event) inside a Segment
-
-signal channels  are handled by group of  "stream".
-one stream will at neo.io level one AnalogSignal with multi-channel.
-
-
-A helper class `neo.io.basefromrawio.BaseFromRaw` transform a RawIO to
+For this level, datasets of recordings are mapped as follows:
+
+A channel refers to a physical channel of recording in an experiment. It is identified by a
+channel_id. Recordings from a channel consist of sections of samples which are recorded
+contiguously in time; in other words, a section of a channel has a specific sampling_rate,
+start_time, and length (and thus also stop_time, which is the time of the sample which would
+lie one sampling interval beyond the last sample present in that section).
+
+A stream consists of a set of channels which all have the same structure of their sections of
+recording and the same data type of samples. Each stream has a unique stream_id and has a name,
+which does not need to be unique. A stream thus has multiple channels which all have the same
+sampling rate and are on the same clock, have the same sections with t_starts and lengths, and
+the same data type for their samples. The samples in a stream can thus be retrieved as an Numpy
+array, a chunk of samples.
+
+Channels within a stream can be accessed be either their channel_id, which must be unique within
+a stream, or by their channel_index, which is a 0 based index to all channels within the stream.
+Note that a single channel of recording may be represented within multiple streams, and such is
+the case for RawIOs which may have both unfiltered and filtered or downsampled versions of the
+signals from a single recording channel. In such a case, a single channel and channel_id may be
+represented by a different channel_index within different streams. Lists of channel_indexes are
+often convenient to pass around a selection of channels within a stream.
```

Comment:
perfect

Reply:
But I still don't understand what are the acceptable types of arguments for get_analog_signal_chunk's channel_indexes argument? Does it include a slice?

Reply:
yes

Reply:
or None or list or np.array int

Reply:
OK, added that clarification to the parameter comment for the function.